### DISCUSSION TITLE
Repeat by completion date

### DISCUSSION THREAD

**Discussion by jojoob** [TechbeeAT/jtxBoard#834] [Ideas]
Some other task apps (eg. TickTick) have a feature to repeat tasks after completion. Like "repeat 2 weeks after completion". This feature is very handy for planning specific tasks and get reminded to do it again after some time.

I don't know the caldav standard and could imagine that such a feature is not included but a client side implementation would totally fine for me. The feature would only have to create a copy of the tasks when marked as completed with updated start/end dates.

**Comment by patrickunterwegs** [comment]
Hey @jojoob , 
currently you can make a recurring entry and set the option to show only one recurring entry in the future. 
This can be handy for use cases like 'bring out the trashes' that recurs every two weeks - a new entry pops up in the future when the current one is done.

What would be a use case for your idea?

**Reply by jojoob** [reply]
Something like 'vacuum the flat'. It may be done Every 5 days but when it's done 3 days late it has not to be done after 2 days again. Instead at best 5 days after last completion.

**Reply by patrickunterwegs** [reply]
Thanks @jojoob !
It's clearer now, but I'm not so convinced that this is such a good idea in the context of jtx Board. The iCalendar standard doesn't support such an option.  It you for example finish a task in another platform, nothing would happen (ie the other platform cannot be made aware of any custom implementation to create a new task when done...)

I'm afraid that this could lead to complaints in jtx Board because people don't understand that this would only be work in jtx Board...

**Reply by jojoob** [reply]
> The iCalendar standard doesn't support such an option.

I already suspected that. :(

> I'm afraid that this could lead to complaints in jtx Board because people don't understand that this would only be work in jtx Board...

When the feature is well explained I think you do not have to fear complaints. And there are already features that only work within jtx Board, namely extended statuses and automatic alarm: always on due.

I hope I can convince you because this feature is really handy for me and at the moment hold be back a little from switching fully to jtx Board. :)

**Reply by bepolymathe** [reply]
Hi,

Obviously, this feature could be useful. But I think you have to take into account the difficulty of maintaining compatibility with the standard, and therefore the applications that use it (thunderbird, evolution, etc). 

From my point of view, this feature has nothing to do with recurrence (which by definition returns at a fixed period) but rather with rescheduling. If, however, you want to keep a record of what's been done, we could imagine a "done and rescheduling" function which involves (1) duplicating the task, (2) assigning it a new due date and (3) checking off the original task. This would not affect use in other applications, and would not affect recurrence logic. 🤔

**Reply by jojoob** [reply]
> If, however, you want to keep a record of what's been done, we could imagine a "done and rescheduling" function which involves (1) duplicating the task, (2) assigning it a new due date and (3) checking off the original task.

I really like that idea! Could you imagine a way of setting a task specific rule (/default time span) for the new due date beforehand? Or would it be necessary to select the date during the "done and rescheduling" action - which would be a bit less practical but better than not having the feature at all.
